### PR TITLE
Remove RTMP streams for nvr connected cams

### PR DESCRIPTION
**Pull request by apocaliss92**

### CODE REVIEW

File: plugins/reolink/src/main.ts [hunk 1]
```diff
@@ -703,21 +703,30 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
         // anecdotally, encoders of type h265 do not have a working RTMP main stream.
         const mainEncType = this.storageSettings.values.abilities?.value?.Ability?.abilityChn?.[rtspChannel].mainEncType?.ver;
 
-        if (live === 2) {
-            if (mainEncType === 1) {
-                streams.push(rtmpSub, rtspMain, rtspSub);
+        const isNvr = this.client.getIsNvr(deviceInfo);
+
+        if (isNvr) {
```
Comment (apocaliss92) [inline plugins/reolink/src/main.ts:708]: Interesting comparing, this is the only block added here (708-719), the non nvr branch is not changed at all

**Comment by koush** [comment]
is there a better way to do this, to query if rtsp is available somehow.

**Comment by apocaliss92** [comment]
Actually the streams are available but are broken when thorough NVR, we guided 2 users already this week to switch from rtmp to rtsp, maybe would just be better to set the defaults to rtsp? In that case what would be the criteria to let Scrypted pick them instead of rtmp, just order in the streams array?

**Comment by koush** [comment]
> Actually the streams are available but are broken when thorough NVR, we guided 2 users already this week to switch from rtmp to rtsp, maybe would just be better to set the defaults to rtsp? In that case what would be the criteria to let Scrypted pick them instead of rtmp, just order in the streams array?

The issue with RTSP is that it is buggy on many reolink cameras causing dropouts or camera crashes. Furthermore RTSP only offers 2 streams and not 3 like RTMP.

**Pull request closed without merging**